x/evm/keeper/grpc_query.go

```diff
@@ -73,66 +76,66 @@ func (q Querier) BurnerInfo(c context.Context, req *types.BurnerInfoRequest) (*t
 	return nil, status.Error(codes.NotFound, "unknown address")
 }
 
-func getExecuteDataAndSigs(ctx sdk.Context, s types.Signer, commandBatch types.CommandBatch) ([]byte, []types.Signature, error) {
-	id := hex.EncodeToString(commandBatch.GetID())
-	sig, sigStatus := s.GetSig(ctx, id)
-	if sigStatus != tss.SigStatus_Signed {
-		return nil, nil, fmt.Errorf("could not find a corresponding signature for sig ID %s", id)
-	}
-
-	switch sig := sig.GetSig().(type) {
-	case *tss.Signature_SingleSig_:
-		commandBatchSig, err := getCommandBatchSig(sig.SingleSig.SigKeyPair, commandBatch.GetSigHash())
-		if err != nil {
-			return nil, nil, err
+func optimizeSignatureSet(sigs map[string]multisigtypes.Signature, addressWeights map[string]sdk.Uint, threshold sdk.Uint) [][]byte {
+	addresses := maps.Keys(addressWeights)
+	sort.SliceStable(addresses, func(i, j int) bool {
+		return addressWeights[addresses[i]].GT(addressWeights[addresses[j]])
+	})
+
+	var optimizedSig [][]byte
+	cumWeight := sdk.ZeroUint()
+	for _, addr := range maps.Keys(addressWeights) {
+		if cumWeight.GTE(threshold) {
+			break
 		}
 
-		executeData, err := types.CreateExecuteDataSinglesig(commandBatch.GetData(), commandBatchSig)
-		if err != nil {
-			return nil, nil, fmt.Errorf("could not create transaction data: %s", err)
+		sig, _ := sigs[addr]
+		if sig != nil {
+			optimizedSig = append(optimizedSig, sig)
+			cumWeight = cumWeight.Add(addressWeights[addr])
 		}
+	}
 
-		return executeData, []types.Signature{commandBatchSig}, nil
-	case *tss.Signature_MultiSig_:
-		sigKeyPairs := types.SigKeyPairs(sig.MultiSig.SigKeyPairs)
-		sort.Stable(sigKeyPairs)
+	return optimizedSig
```

Comment:
Signature needs to be resorted by operator address.

Reply:
😭

Reply:
Maybe make this into an optimized address => sig map. And then can convert it to a slice by mapping over the operator list in correct order, and adding a signature for that operator if applicable?